src/cargo/sources/git/utils.rs

```diff
@@ -97,10 +97,21 @@ impl GitRemote {
         reference: &GitReference,
         cargo_config: &Config,
     ) -> CargoResult<(GitDatabase, GitRevision)> {
+        let format_error = |operation| {
+            let uses_cli =  cargo_config.net_config().ok()
+                .and_then(|config| config.git_fetch_with_cli)
+                .unwrap_or(false);
+            if !uses_cli {
+                format!("failed to {} into '{}'. If your environment requires git authentication or proxying, try enabling `[net] git-fetch-with-cli = true` in `.cargo/config`", operation, into.display())
```

Comment:
I think it would be good to avoid writing this as `[net] git-fetch-with-cli = true`, since pasting that as-is results in a (very old) giant warning, and is not valid TOML.  Maybe just `net.git-fetch-with-cli = true`? Or display it in separate lines, like:

```
    [net]
    git-fetch-with-cli = true
```

Also, I think it would be good to wrap this line. rustfmt doesn't handle large strings automatically, so a backslash-newline somewhere in the string will help it along.

Reply:
I've changed it to point to the docs instead.